lib/tether/ops_linux.go

```diff
@@ -326,6 +328,37 @@ func renameLink(t Netlink, link netlink.Link, slot int32, endpoint *NetworkEndpo
 	return link, nil
 }
 
+func gratuitous(t Netlink, link netlink.Link, endpoint *NetworkEndpoint) error {
+	if ip.IsUnspecifiedIP(endpoint.Assigned.IP) {
+		// refusing to broadcast unspecified IP
+		return nil
+	}
+
+	intf, err := net.InterfaceByIndex(link.Attrs().Index)
+	if err != nil {
+		return err
+	}
+
+	client, err := arp.Dial(intf)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		client.Close()
+	}()
+
+	srcHW := link.Attrs().HardwareAddr
+	srcPA := endpoint.Assigned.IP
+
+	gratuitousPacket, err := arp.NewPacket(arp.OperationRequest, srcHW, srcPA, ethernet.Broadcast, srcPA)
```

Comment:
What if you get a response for this ARP request? Which of course means IP address conflict.
I think we may want to send this ARP request a few times just in case of packet loss or some congestion happens and packet is accidentally was dropped.

I would run a goroutine that sends the same packet twice after that later.

Reply:
@vburenin I'm going to stick with just the one which is in line with the linux networking stack when it issues them. In a distributed switch vSphere will be doing ARP suppression and processing anyway  to avoid ARP cascades. This is primarily an issue for us in the bridge networks as that's where we get very rapid reuse of the same IP addresses and in that case it's the DVS/vSphere level caches we care about.